src/test/java/com/exadel/discountwebapp/role/RoleServiceIntegrationTest.java

```diff
@@ -0,0 +1,67 @@
+package com.exadel.discountwebapp.role;
+
+import com.exadel.discountwebapp.DiscountWebApplication;
+import com.exadel.discountwebapp.role.entity.Role;
+import com.exadel.discountwebapp.role.mapper.RoleMapper;
+import com.exadel.discountwebapp.role.repository.RoleRepository;
+import com.exadel.discountwebapp.role.service.RoleService;
+import com.exadel.discountwebapp.role.vo.RoleResponseVO;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.transaction.support.TransactionTemplate;
+import java.util.List;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Sql(executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, scripts =
+        "classpath:cleanup.sql")
+@SpringBootTest(classes = DiscountWebApplication.class)
+public class RoleServiceIntegrationTest {
+
+    @Autowired
+    private RoleMapper roleMapper;
+    @Autowired
+    private RoleRepository roleRepository;
+    @Autowired
+    private RoleService roleService;
+    @Autowired
+    private TransactionTemplate transactiontemplate;
+
+    private Role roleUser;
+    private Role roleAdmin;
+
+    @BeforeEach
+    void setUp() {
+        transactiontemplate.execute(status -> {
+            roleUser = Role.builder()
```

Comment:
creating roles can be done outside the transaction scope
a transaction should be kept as short as possible